### PR TITLE
Remove TryAllowed from scope checker

### PR DESCRIPTION
**Pull request by janisz**
## Description

There is no need to try with only built in authorizer and no error scope checker.

## Testing Performed

CI

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2726** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2726" target="_blank">Graphite</a> 
    * **PR #2722** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2722" target="_blank">Graphite</a> 
      * **PR #2750** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2750" target="_blank">Graphite</a> 
        * **PR #2751** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2751" target="_blank">Graphite</a> 
          * **PR #2752** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2752" target="_blank">Graphite</a> 
            * **PR #2774** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2774" target="_blank">Graphite</a> 
              * **PR #2775** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2775" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2775?utm_source=stack-comment).

File: central/compliance/datastore/sac_filter.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (ds *sacFilterImpl) filterClusters(ctx context.Context, clusters set.String
 	// Filter the compliance results by cluster.
 	allowed := set.NewStringSet()
 	for cluster := range clusters {
-		if resourceScopeChecker.TryAllowed(sac.ClusterScopeKey(cluster)) {
+		if ok, _ := resourceScopeChecker.Allowed(ctx, sac.ClusterScopeKey(cluster)); ok {
```
Comment (rhybrillou) [inline central/compliance/datastore/sac_filter.go:110]: Would it make sense to propagate the potential error from the Allowed function ? Or does `Allowed` always return `nil` anyway and is being re-factored in a later pull request ?

File: pkg/sac/scope_checker.go [hunk 2]
```diff
@@ -14,7 +14,6 @@ import (
 //go:generate mockgen-wrapper
 type ScopeChecker interface {
 	SubScopeChecker(keys ...ScopeKey) ScopeChecker
-	TryAllowed(subScopeKeys ...ScopeKey) bool
```
Comment (rhybrillou) [inline pkg/sac/scope_checker.go:17]: Shouldn't the function implementation (lines 58 to 66) be removed as well ?

File: pkg/sac/scope_key.go [hunk 3]
```diff
@@ -159,7 +160,8 @@ type ScopeSuffix []ScopeKey
 
 // TryAllowed implements the ScopePredicate interface.
 func (i ScopeSuffix) TryAllowed(sc ScopeChecker) bool {
```
Comment (rhybrillou) [inline pkg/sac/scope_key.go:162]: Would it make sense to rename the `ScopePredicate` `TryAllowed` function to `Allowed` or something similar as the two-pass logic `TryAllowed` / `PerformChecks` / `TryAllowed` is not needed anymore ?
This is candidate for a follow-up pull request if not already planned.

File: central/compliance/datastore/sac_filter.go (hunk 1 above)
Comment (janisz) [reply central/compliance/datastore/sac_filter.go:110]: It always returns `nil` and will be refactored in https://github.com/stackrox/stackrox/pull/2780

File: pkg/sac/scope_checker.go (hunk 2 above)
Comment (janisz) [reply pkg/sac/scope_checker.go:17]: You're right it should be removed here not in https://github.com/stackrox/stackrox/pull/2778/files

Comment (janisz) [reply pkg/sac/scope_checker.go:17]: But it's used by other functions so let's keep it here for now

File: pkg/sac/scope_key.go (hunk 3 above)
Comment (janisz) [reply pkg/sac/scope_key.go:162]: It's done here https://github.com/stackrox/stackrox/pull/2779

File: pkg/sac/scope_checker.go (hunk 2 above)
Comment (rhybrillou) [reply pkg/sac/scope_checker.go:17]: ok, fine :)

**Review by rhybrillou — APPROVED** [review]

**Pull request merged**